src/Bicep.Core/Syntax/ImportedSymbolsListItemSyntax.cs

```diff
@@ -1,25 +1,40 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
+using System;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Extensions;
 using Bicep.Core.Navigation;
 using Bicep.Core.Parsing;
 
 namespace Bicep.Core.Syntax;
 
 public class ImportedSymbolsListItemSyntax : SyntaxBase, INamedDeclarationSyntax
 {
-    public ImportedSymbolsListItemSyntax(IdentifierSyntax originalSymbolName, SyntaxBase? asClause)
+    public ImportedSymbolsListItemSyntax(SyntaxBase originalSymbolName, SyntaxBase? asClause)
     {
+        AssertSyntaxType(originalSymbolName, nameof(originalSymbolName), typeof(StringSyntax), typeof(IdentifierSyntax), typeof(SkippedTriviaSyntax));
```

Comment:
What's the use case for supporting a string in the `originalSymbolName` field?

Reply:
Names in ARM templates don't have to be valid Bicep identifiers. You can have and use a variable named `'@!$^%#`, and there are only a few ways we could deal with that when importing from a compiled template:

1. Raise a diagnostic on the `import` statement indicating that the imported template has issues, even though it doesn't violate any of ARM's rules.
2. Ignore any import whose name isn't a valid identifier in Bicep.
3. Allow `import` statements to use a string for the original symbol name but require that such imports alias the symbol to a valid identifier (e.g., `import {'a-b' as ab} from 'template.json'`.

I didn't feel great about option 2 because it silently drops exported symbols without explaining why to the user. Option 1 has a couple problems that made me think it's not a viable solution:

1. There's no guarantee that the author of the template with the `import` statement can edit the imported artifact (they may be importing a template spec or registry module to which they have read privileges only).
2. The user will have to search the imported template for the problem, since the diagnostic will be associated with the `import` statement in the *consuming* template rather than the syntax that *exports* the symbol from the offending template.

I didn't see any drawbacks to option 3 besides some additional complexity in the parser. Similar syntax is supported in [js imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#syntax), so there was a good example to follow.